### DISCUSSION TITLE
[分享] 一个 Memos Twitter bot

### DISCUSSION THREAD

**Discussion by Ice-Hazymoon** [usememos/memos#1182] [Show and tell]
https://github.com/Ice-Hazymoon/memos-twitter-bot

可以实现将推文私信给bot，或者在某条推文下 at bot 就能将推文的信息和图片视频保存到 memos 中，代码很热乎，bug 也很多，受 Twitter API 的局限性（申请难，实时处理私信要企业权限，需要单独开一个 bot 账号，api 速率限制等问题），只能自己部署。

一开始是想用 go 实现的，但由于 go 的 twitter sdk 已经停止维护，bug 也一大堆，不想自己写，所以就用万能的 js 了。

我觉得没有人会用这么麻烦的东西，所以就没写文档，代码很简单，有需要的可以自己搭建。

![image](https://user-images.githubusercontent.com/25239977/221563178-d0d8e853-0882-4279-9567-82feea783b45.png)
![image](https://user-images.githubusercontent.com/25239977/221563132-a4bcbf31-18af-4fae-988a-45d90e6b9788.png)
![image](https://user-images.githubusercontent.com/25239977/221563155-e1725f74-73f5-45ba-9fb7-fc88b908fe4e.png)

**Comment by skyue** [comment]
可以进一步考虑下反向需求，把memos当twitter客户端用，把打上特定标签的memos同步到Twitter。

**Reply by love2wind** [reply]
> 可以进一步考虑下反向需求，把memos当twitter客户端用，把打上特定标签的memos同步到Twitter。

不错的想法

**Comment by wangyunzi** [comment]
太棒的想法了

**Comment by llity** [comment]
希望有更友好的方法实现

**Comment by woshichenghaibo** [comment]
不错的
